### DISCUSSION TITLE
Base class for HydroComponent

### DISCUSSION THREAD

**Discussion by taublast** [hydrostack/hydro#38] [Ideas]
Hydro went by subclassing `ViewComponent`, to stand out completely from Blazor thus cutting compatibility with existing Blazor libs/projects/techniques.
Is it theoretically possible to hook inside Blazor infrastructure, still use LayoutComponentBase/ComponentBase, and just replace out-of-the-box Blazor behavior with Hydro? For example, if we omit (net8 case) use `app.MapRazorComponents<App>() .AddInteractiveServerRenderMode();` and use Hydro instead, could Hydro take over in theory by implementing a different infrastructure?

 `app.UseBlazorHydro();` a dream?

**Comment by kjeske** [comment]
Hi! I tried that idea really hard in the beginning. I wanted to use the Razor components model (.razor files) and attach my custom logic, but the implementation of Blazor was quite locked (many internals), so it wasn't possible.
